gazelle_haskell_modules/rule_generation.go

```diff
@@ -119,15 +119,19 @@ func haskellModuleRulesToRuleInfos(
 		}
 
 		modDatas := haskellModulesToModuleData([]string{src})
-		ruleInfo := RuleInfo{
-			OriginatingRules: oRules,
-			ModuleData:       modDatas[0],
-		}
+		if len(modDatas) > 0 {
+			ruleInfo := RuleInfo{
+				OriginatingRules: oRules,
+				ModuleData:       modDatas[0],
+			}
 
-		ruleInfoss = append(ruleInfoss, []*RuleInfo{&ruleInfo})
+			ruleInfoss = append(ruleInfoss, []*RuleInfo{&ruleInfo})
 
-		r.SetPrivateAttr(PRIVATE_ATTR_MODULE_NAME, ruleInfo.ModuleData.ModuleName)
-		r.SetPrivateAttr(PRIVATE_ATTR_ORIGINATING_RULE, ruleInfo.OriginatingRules)
+			r.SetPrivateAttr(PRIVATE_ATTR_MODULE_NAME, ruleInfo.ModuleData.ModuleName)
+			r.SetPrivateAttr(PRIVATE_ATTR_ORIGINATING_RULE, ruleInfo.OriginatingRules)
+		} else {
+			log.Printf("warning: found no module source for haskell_module rule %s: %q\n", r.Name(), src)
```

Comment:
I enhanced this message in a49dd86. I finally didn't add a suggestion to use the fix command to reduce the noise in the output and keep the code simpler. This message as it stands is printed both when fixing and when not fixing.